### PR TITLE
immer fs final

### PR DESCRIPTION
**Pull request by songgao**

### CODE REVIEW

File: shared/fs/banner/system-file-manager-integration-banner/index.tsx [hunk 1]
```diff
@@ -180,7 +180,7 @@ const Enabled = (props: Props) => {
 type JustEnabledProps = {onDismiss: null | (() => void)}
 const JustEnabled = ({onDismiss}: JustEnabledProps) => {
   const preferredMountDirs = Container.useSelector(state => state.fs.sfmi.preferredMountDirs)
-  const displayingMountDir = preferredMountDirs.get(0) || ''
+  const displayingMountDir = preferredMountDirs[0] || ''
```
Comment (jakob223) [inline shared/fs/banner/system-file-manager-integration-banner/index.tsx]: Do you still need the `|| ''`? If you do, can this crash if `preferredMountDirs` is empty?

File: shared/fs/common/path-info.tsx [hunk 2]
```diff
@@ -16,7 +16,7 @@ const useMountPointPath = (platformAfterMountPath: string) => {
   const sfmi = Container.useSelector(state => state.fs.sfmi)
   const mount =
     sfmi.driverStatus.type === Types.DriverStatusType.Enabled
-      ? sfmi.preferredMountDirs.get(0) || sfmi.directMountDir
+      ? sfmi.preferredMountDirs[0] || sfmi.directMountDir
```
Comment (jakob223) [inline shared/fs/common/path-info.tsx]: same comment as above

File: shared/stories/__tests__/__snapshots__/Storyshots.test.js.snap [hunk 3]
```diff
@@ -260150,96 +260150,6 @@ Array [
               >
                 Keybase is enabled in your Finder.
               </span>
-              <div
```
Comment (jakob223) [inline shared/stories/__tests__/__snapshots__/Storyshots.test.js.snap]: have you made sure this story change is intentional?

File: shared/fs/banner/system-file-manager-integration-banner/index.tsx (hunk 1 above)
Comment (songgao) [reply shared/fs/banner/system-file-manager-integration-banner/index.tsx]: I think we do, if not for actual empty array, then at least TSC. If the array is empty, `preferredMountDir[0]` would be `undefined` so it won't crash.

File: shared/stories/__tests__/__snapshots__/Storyshots.test.js.snap (hunk 3 above)
Comment (songgao) [reply shared/stories/__tests__/__snapshots__/Storyshots.test.js.snap]: I remember noticing this too, but checked all relative changes and thought it was fine so I just left it assuming master storybook inconsistency. I just checked again, and guess what ... I forgot to update the storybook prop-providers to de-immutable-ize! Thanks for calling it out.

**Pull request merged**